From 713dad73647e6b4f55cce9eb230d6f43f4567500 Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Mon, 13 Feb 2017 16:24:36 +0800
Subject: [PATCH 1524/5242] MLK-13914-03 ARM: dts: Add cpu setpoints and
 clocks properties on i.mx7ulp

commit  74dec2b60ee4169fba1b14089b3343183e57527c from
https://source.codeaurora.org/external/imx/linux-imx.git

Add CPU setpoints property on i.MX7ULP A0 part, the setpoints table is a preliminary
one, will update it according to the datasheet when the final one is available.

The setpoints we can currently used is as below:
	416072 KHz/0.925V,
	531648 KHz/1.025V;

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk.dts |    5 +++++
 arch/arm/boot/dts/imx7ulp.dtsi    |   17 +++++++++++++++++
 2 files changed, 22 insertions(+)

diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index 493c6de..9014aa4 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -1,5 +1,6 @@
 /*
  * Copyright 2016 Freescale Semiconductor, Inc.
+ * Copyright 2017 NXP.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
@@ -181,6 +182,10 @@
 	};
 };
 
+&cpu0 {
+	arm-supply= <&sw1_reg>;
+};
+
 &iomuxc1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_hog_1>;
diff --git a/arch/arm/boot/dts/imx7ulp.dtsi b/arch/arm/boot/dts/imx7ulp.dtsi
index 6ec834b..dc7e8ef 100644
--- a/arch/arm/boot/dts/imx7ulp.dtsi
+++ b/arch/arm/boot/dts/imx7ulp.dtsi
@@ -37,7 +37,24 @@
 		cpu0: cpu@0 {
 			compatible = "arm,cortex-a7";
 			device_type = "cpu";
+			clock-latency = <61036>; /* two CLK32 periods */
 			reg = <0>;
+
+		operating-points = <
+			/* KHz	uV */
+			531648	1025000
+			416072	925000
+		>;
+		clocks = <&clks IMX7ULP_CLK_ARM>,
+			 <&clks IMX7ULP_CLK_CORE_DIV>,
+			 <&clks IMX7ULP_CLK_SYS_SEL>,
+			 <&clks IMX7ULP_CLK_HSRUN_SYS_SEL>,
+			 <&clks IMX7ULP_CLK_HSRUN_CORE>,
+			 <&clks IMX7ULP_CLK_SPLL_PFD0>,
+			 <&clks IMX7ULP_CLK_SPLL_SEL>,
+			 <&clks IMX7ULP_CLK_FIRC>;
+		clock-names = "arm", "core_div", "sys_sel", "hsrun_sys_sel",
+			      "hsrun_core", "spll_pfd0", "spll_sel", "firc";
 		};
 	};
 
-- 
1.7.9.5

